java-checks/src/main/java/org/sonar/java/checks/xml/web/AbstractWebXmlXPathBasedCheck.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.xml.web;
+
+import org.sonar.java.checks.xml.AbstractXPathBasedCheck;
+import org.sonarsource.analyzer.commons.xml.XmlFile;
+
+public abstract class AbstractWebXmlXPathBasedCheck extends AbstractXPathBasedCheck {
```

Comment:
This is a first step. I think we could have this logic a bit more wider, to run some checks only when matching a given file name, or filename predicate. WDYT?

This would allow to simplify checks targeting `pom.xml` files, and maybe others.

In the meantime, I think it's OK to merge this already.

Reply:
Yes, I totally agree with that.